File: app/src/main/java/com/example/mateusz/coffeenotes/BeansTypeDataManager.java
package com.example.mateusz.coffeenotes;

import java.util.ArrayList;
import java.util.List;
import java.util.UUID;

public class BeansTypeDataManager {
  private static BeansTypeDataManager instance;

  private List<BeansType> beansTypeList;

  private BeansTypeDataManager() {
    beansTypeList = new ArrayList<>();

    for (int i = 0; i < 10; ++i) {
      BeansType beansType = new BeansType();
      beansType.setName("Type #" + Integer.toString(i));
      beansType.setCountry("Country #" + Integer.toString(i));
      beansTypeList.add(beansType);
    }
  }

  public static BeansTypeDataManager getInstance() {
    if (instance == null) {
      instance = new BeansTypeDataManager();
    }
    return instance;
  }

  public BeansType getBeansTypeById(UUID id) {
    for (BeansType beansType : beansTypeList) {
      if (beansType.getId().equals(id)) {
        return beansType;
      }
    }
    return null;
  }

  public List<BeansType> getBeansTypeList() {
    return beansTypeList;
  }
}

File: app/src/main/java/com/example/mateusz/coffeenotes/database/BeansTypeDataManager.kt
package com.example.mateusz.coffeenotes.database

import com.example.mateusz.coffeenotes.BeansType
import com.example.mateusz.coffeenotes.CoffeeNote
import java.io.File
import java.util.UUID

interface BeansTypeDataManager {
    fun getCoffeeNotes(): List<CoffeeNote>

    fun getCoffeeNoteById(id: UUID): CoffeeNote?

    fun saveCoffeeNote(coffeeNote: CoffeeNote)

    fun removeCoffeeNote(coffeeNote: CoffeeNote)

    fun getBeansTypes(): List<BeansType>

    fun getBeansTypeById(id: UUID): BeansType?

    fun saveBeansType(beansType: BeansType)

    fun removeBeansType(beansType: BeansType)

    fun getPhotoFile(beansType: BeansType): File
}